### PR TITLE
Removed conditional_agents and agent_count parameters from initialize…

### PR DESCRIPTION
**Pull request by KieranRatcliffeInvertedAI**
…_request

### CODE REVIEW

File: invertedai_cpp/invertedai/initialize_request.cc [hunk 1]
```diff
@@ -79,8 +53,8 @@ InitializeRequest::InitializeRequest(const std::string &body_str) {
   this->get_infractions_ = this->body_json_["get_infractions"].is_boolean()
     ? this->body_json_["get_infractions"].get<bool>()
     : false;
-  this->agent_count_ = this->body_json_["agent_count"].is_number_integer()
-    ? std::optional<int>{this->body_json_["agent_count"].get<int>()}
+  this->num_agents_to_spawn_ = this->body_json_["num_agents_to_spawn"].is_number_integer()
+    ? std::optional<int>{this->body_json_["num_agents_to_spawn"].get<int>()}
```
Comment (rf-ivtdai) [inline invertedai_cpp/invertedai/initialize_request.cc:57]: Thanks! As testing for this change, could you please add a dedicated test flow to initialize where num_agent_to_spawn is not set,  and both agent attributes and state history has a valid, positive size?

Comment (KieranRatcliffeInvertedAI) [reply invertedai_cpp/invertedai/initialize_request.cc:57]: While I think these tests are valuable, do you think we should make that into a separate PR?

Comment (rf-ivtdai) [reply invertedai_cpp/invertedai/initialize_request.cc:57]: Sure :)

Comment (KieranRatcliffeInvertedAI) [reply invertedai_cpp/invertedai/initialize_request.cc:57]: Is the PR acceptable otherwise?

Comment (rf-ivtdai) [reply invertedai_cpp/invertedai/initialize_request.cc:57]: As long as relevant testing will be added in the near future, yes!

**Pull request merged**